antarest/core/maintenance/service.py

```diff
@@ -0,0 +1,194 @@
+import logging
+from typing import Optional, Callable
+
+from fastapi import HTTPException
+
+from antarest.core.config import Config
+from antarest.core.configdata.model import ConfigDataAppKeys
+from antarest.core.interfaces.cache import ICache
+from antarest.core.interfaces.eventbus import (
+    IEventBus,
+    EventType,
+    Event,
+)
+from antarest.core.maintenance.model import (
+    MaintenanceMode,
+)
+from antarest.core.maintenance.repository import MaintenanceRepository
+from antarest.core.model import PermissionInfo, PublicMode
+from antarest.core.requests import (
+    RequestParameters,
+    MustBeAuthenticatedError,
+    UserHasNotPermissionError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class MaintenanceService:
+    def __init__(
+        self,
+        config: Config,
+        repository: MaintenanceRepository,
+        event_bus: IEventBus,
+        cache: ICache,
+    ):
+        self.config = config
+        self.repo = repository
+        self.event_bus = event_bus
+        self.cache = cache
+
+    def _get_maintenance_data(
+        self,
+        cache_id: str,
+        db_call: Callable[[int], Optional[str]],
+        json_key: str,
+        cache_get_error: str,
+        database_get_error: str,
+        database_not_found: str,
+        raise_db_not_found: bool,
+        cache_set_error: str,
+        default_value: str,
+        request_params: RequestParameters,
+    ) -> str:
+
+        if not request_params.user:
+            raise MustBeAuthenticatedError()
+
+        # Get maintenance status from cache
+        try:
+            data_json = self.cache.get(cache_id)
+            if data_json is not None and json_key in data_json.keys():
+                return str(data_json[json_key])
+        except Exception as e:
+            logger.error(cache_get_error, exc_info=e)
+
+        # Else get from database
+        try:
+            data: Optional[str] = db_call(0)
+        except Exception as e:
+            logger.error(database_get_error, exc_info=e)
```

Comment:
Use a simple log like "Failed to fetch {cache_id} from database" (even if it's not the thing we fetch from database, it's sufficient for debug / log analysis purpose)